R/compat-obj-type.R

```diff
@@ -74,7 +80,12 @@ obj_type_friendly <- function(x, value = TRUE, length = FALSE) {
         typeof(x),
         logical = if (x) "`TRUE`" else "`FALSE`",
         integer = "an integer",
-        double = "a number",
+        double =
+          if (is.infinite(x)) {
+            "an infinite number"
```

Comment:
I actually like seeing ``"`Inf`"`` and ``"`-Inf`"`` rather than this, if that sounds good to you. Since we also like `NA` and `NaN` out explicitly